v8py/jsfunction.cpp

```diff
@@ -24,6 +27,12 @@ PyObject *js_function_call(js_function *self, PyObject *args, PyObject *kwargs)
     IN_CONTEXT(self->context.Get(isolate));
     JS_TRY
 
+    double timeout = 0;
+    {
+        context_c *ctx_c = (context_c *) context->GetEmbedderData(CONTEXT_OBJECT_SLOT).As<External>()->Value();
+        timeout = ctx_c->timeout;
+    }
```

Comment:
This chunk of code is duplicated in quite a few places. The only place the timeout argument to setup_timeout is not always the context's timeout is in Context.eval. So why not create a new function that gets the context's timeout and passes that to setup_timeout, and use that everywhere but Context.eval? And do something similar for cleanup_timeout, of course.